### PR TITLE
Mbs 8094 fetch performance results

### PR DESCRIPTION
**Pull request by dsvoronin**
First naive implementation, without fallbacks to older commits

### CODE REVIEW

File: subprojects/gradle/performance/src/main/kotlin/com/avito/performance/PerformanceExtension.kt [hunk 1]
```diff
@@ -25,4 +26,21 @@ open class PerformanceExtension @Inject constructor(objects: ObjectFactory) {
...
+     */
+    @Suppress("UnstableApiUsage")
+    val targetBranchResultSource: Property<TargetBranchResultSource> =
+        objects.property<TargetBranchResultSource>()
+            .convention(TargetBranchResultSource.RUN_IN_PROCESS)
+
+    /**
+     * FETCH_FROM_OTHER_BUILD strategy needs to know about build config id
+     */
+    val targetBuildConfigId = objects.property<String>()
```
Comment (eugene-krivobokov) [inline subprojects/gradle/performance/src/main/kotlin/com/avito/performance/PerformanceExtension.kt]: Can you move this parameter into FETCH_FROM_OTHER_BUILD strategy dsl?

```
build.gradle:

performance {
     targetBranchResultSource = runInProcess()
}

performance {
     targetBranchResultSource = fetchFromOtherBuild(id = "build id")
}
```
Flat parameters have more inconsistent states. It's harder to understand and validate

Great example is [test filter](https://avito-tech.github.io/avito-android/docs/test/runner/#filtering-tests-for-execution)

Comment (dsvoronin) [reply subprojects/gradle/performance/src/main/kotlin/com/avito/performance/PerformanceExtension.kt]: will do; was afraid of groovy/kotlin compatibility

**Comment by dsvoronin** [comment]
TODO: disable rerun in Instrumentation plugin

File: subprojects/gradle/performance/src/main/kotlin/com/avito/performance/PerformancePlugin.kt [hunk 2]
```diff
@@ -112,29 +115,58 @@ open class PerformancePlugin : Plugin<Project> {
...
+                                    }
+                                    FETCH_FROM_OTHER_BUILD -> {
+
+                                        //there is no need to depend on tests, it is a hack to postpone fetching
+                                        dependsOn(sourceBranchResultsCollector)
+
+                                        //todo probably it's a mistake; but we need to fetch teamcity API otherwise
+                                        buildId.set(null as String?)
+
+                                        this.reportCoordinates.set(
```
Comment (sergioboi) [inline subprojects/gradle/performance/src/main/kotlin/com/avito/performance/PerformancePlugin.kt:157]: I didn't get how it works now.
We have develop build produces the target data
And have PR build that consumes the target data
So this logic means we will consume target data from `originalBranch.commit` `extension.targetBuildConfigId`
What are the `originalBranch.commit` and `extension.targetBuildConfigId`?
Who, where, what and when did set `extension.targetBuildConfigId`?

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/gradle/performance/src/main/kotlin/com/avito/performance/PerformancePlugin.kt (hunk 2 above)
Comment (dsvoronin) [reply subprojects/gradle/performance/src/main/kotlin/com/avito/performance/PerformancePlugin.kt:157]: it was a mistake, i needed `targetBranch.commit`, not a `originalBranch.commit`
also `targetBuildConfigId` intended to be set from extension in application's `build.gradle`

**Pull request merged**